src/NuGet.Core/NuGet.ProjectModel/PackageSpecUtility.cs

```diff
@@ -101,9 +114,23 @@ public static bool IsSnapshotVersion(string version)
                 throw new ArgumentNullException(nameof(projectFramework));
             }
 
+            var hasImports = imports?.Any() == true;
             var hasATF = assetTargetFallback?.Any() == true;
             var hasPTF = packageTargetFallback?.Any() == true;
 
+            // Legacy Project.Json case
+            if (hasImports && !hasATF && !hasPTF)
+            {
+                if (isAssetTargetFallback)
```

Comment:
why this flag `isAssetTargetFallback` matter for project.json scenario? How it's being used for pj?